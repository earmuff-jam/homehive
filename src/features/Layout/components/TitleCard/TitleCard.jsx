import React from "react";

import { Box, Card, CardContent, Chip, Typography } from "@mui/material";

export default function TitleCard({
  title,
  subtitle,
  icon,
  chipLabels,
  onClick = () => {},
}) {
  return (
    <Card
    elevation={0}
    onClick={onClick}
      sx={{
        cursor: "pointer",
        transition: "transform 0.2s",
        "&:hover": {
          transform: "translateY(-4px)",
        },
      }}
    >
      <CardContent sx={{ p: 4, textAlign: "center" }}>
        <Box
          sx={{
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            mb: 2,
          }}
        >
          {icon}
          <Typography variant="h4">{title}</Typography>
        </Box>
        <Typography variant="body1" color="textSecondary" gutterBottom>
          {subtitle}
        </Typography>
        <Box
          sx={{
            display: "flex",
            gap: 1,
            justifyContent: "center",
            flexWrap: "wrap",
          }}
        >
          {chipLabels?.map((label) => (
            <Chip key={label} label={label} size="small" />
          ))}
        </Box>
      </CardContent>
    </Card>
  );
}
