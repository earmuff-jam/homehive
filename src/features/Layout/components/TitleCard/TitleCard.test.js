import React from "react";

import TitleCard from "./TitleCard";
import { Typography } from "@mui/material";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

describe("TitleCard", () => {
  const mockProps = {
    title: "Test Title",
    subtitle: "This is a subtitle",
    icon: <Typography data-testid="test-icon">ðŸ””</Typography>,
    chipLabels: ["Chip 1", "Chip 2"],
  };

  it("renders title, subtitle, icon, and chips", () => {
    render(<TitleCard {...mockProps} />);

    expect(screen.getByText("Test Title")).toBeInTheDocument();
    expect(screen.getByText("This is a subtitle")).toBeInTheDocument();

    expect(screen.getByTestId("test-icon")).toBeInTheDocument();
    expect(screen.getByText("Chip 1")).toBeInTheDocument();
    expect(screen.getByText("Chip 2")).toBeInTheDocument();
  });

  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<TitleCard />);
    expect(asFragment()).toMatchSnapshot();
  });
});
