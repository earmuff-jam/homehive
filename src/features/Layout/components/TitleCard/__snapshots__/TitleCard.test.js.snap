// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleCard renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1f1notb-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiCardContent-root css-1casv0q-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-1s00k8g"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
        />
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-91oxh-MuiTypography-root"
      />
      <div
        class="MuiBox-root css-ocwszi"
      />
    </div>
  </div>
</DocumentFragment>
`;
